### DISCUSSION TITLE
Identifying a minimum of several dates in study definition

### DISCUSSION THREAD

**Discussion by annaschultze** [opensafely/documentation#274] [Q&A] [answered]
I understand a fix for this was recently implemented (https://github.com/opensafely-core/cohort-extractor/pull/571), I've just attempted to use this but I'm not sure if I'm specifying it correctly. Code and error message below. 

```
    any_vte_gp=patients.with_these_clinical_events(
        vte_codes_primary_care,
        returning="date", 
        date_format="YYYY-MM-DD",
        on_or_after="index_date",
        find_first_match_in_period=True,
        return_expectations={"date": {"earliest": "index_date"}},
    ),
    any_vte_hospital=patients.admitted_to_hospital(
        returning="date_admitted",
        with_these_diagnoses=vte_codes_secondary_care,
        on_or_after="index_date",
        date_format="YYYY-MM-DD",
        find_first_match_in_period=True,
        return_expectations={"date": {"earliest": "index_date"}},
    ), 
    any_vte_death=patients.with_these_codes_on_death_certificate(
       vte_codes_secondary_care,
       on_or_after="index_date",
       match_only_underlying_cause=True,
       returning="date_of_death", 
       date_format="YYYY-MM-DD",
       return_expectations={"date": {"earliest": "index_date"}},
    ), 
    any_vte=patients.minimum_of("any_vte_gp", "any_vte_hospital", "any_vte_death"),
```

![image](https://user-images.githubusercontent.com/59482746/122207199-c5df5c80-ce99-11eb-8ee6-2c4ba658f2e0.png)

**Comment by wjchulme** [comment]
~You should drop the `return_expectations` arguments. The variable is always calculated from other variables, whether in dummy data or real data, so expectations aren't needed, and using them throws an error~

Oh hang on, sorry, there is no `return_expectations` argument in the `minimum_of` call

**Comment by alexwalkerepi** [comment, marked as answer]
Is it locally that you're getting an error ? If so have you done an `opensafely upgrade` and `opensafely pull` recently?

**Reply by annaschultze** [reply]
Ah, that solved it! I'd run `opensafely upgrade` but had forgotten to also run `opensafely pull`.
